### DISCUSSION TITLE
Domoticz MQTT last seen update

### DISCUSSION THREAD

**Discussion by bonzi9** [arendst/Tasmota#16836] [Q&A] [answered]
As I am fairly new to Tasmota and MQTT, I have a question about expected behavior.

I have two devices running Tasmota 12.1.1, connected to Mosquitto MQTT broker running on the same system as Domoticz, they seem to function fine in Domoticz after configuring dummy devices and setting the idx in Tasmota.
Checking the logs, one thing I noticed is the device containing a sensor (NodeMCU with DHT22) sends the Domoticz specific message every teleperiod, along with the standard 'STATE' and 'SENSOR' messages: 
`domoticz/in = {"idx":7,"nvalue":0,"svalue":"18.3;64.8;1","Battery":100,"RSSI":6}`
This is what seems to update the last seen status in Domoticz.

The other device is a wall socket (Sonoff S26), and it only sends the Domoticz specific message after boot once:
`domoticz/in = {"idx":8,"nvalue":1,"svalue":"","Battery":100,"RSSI":5}`
This seems to cause the last seen time not to change when a only a new 'STATE' message is sent every teleperiod. However the last seen time does update when the switch is toggled in either Domoticz, Tasmota or a physical button press.
At first I thought the connection of the Sonoff was acting up because it didn't update the last seen time every teleperiod (300 seconds).

Is this the expected behavior difference between a sensor and a switch? Because then I can just interpret the last seen time of the switch as 'last toggled'.

**Comment by arendst** [comment, marked as answer]
Yes.
